bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ChannelItemProvider.java

```diff
@@ -123,13 +127,21 @@ protected void unsetItemChannelLinkRegistry(ItemChannelLinkRegistry linkRegistry
     }
 
     protected void activate(Map<String, Object> properties) {
+        modified(properties);
+    }
+
+    protected synchronized void modified(Map<String, Object> properties) {
         if (properties != null) {
             String enabled = (String) properties.get("enabled");
             if ("false".equalsIgnoreCase(enabled)) {
                 this.enabled = false;
+            } else if ("true".equalsIgnoreCase(enabled)) {
```

Comment:
shouldn't this be else (without if)?
If the value is xxx, the default should be used (which is true) - then it would be the same behavior independent of which state the service was in before.